File: interviewbit-binary-search-sorted-insert-position.java
public class Solution {
    public int searchInsert(ArrayList<Integer> a, int b) {
        int start = 0;
        int end = a.size() - 1;
        int mid;
        int val;
        while (start <= end) {
            mid = (start + end) / 2;
            val = a.get(mid);
            if (val == b) {
                return mid;
            } else if (val < b) {
                start++;
            } else {
                end--;
            }
        }
        return start;
    }
}

File: interviewbit-binary-search-sorted-insert-position.py
class Solution:
    # @param A : list of integers
    # @param B : integer
    # @return an integer
    def searchInsert(self, A, B):
        start = 0
        end = len(A) - 1
        while start <= end:
            mid = (start + end) / 2
            if A[mid] == B:
                return mid
            elif A[mid] < B:
                start = mid + 1
            else:
                end = mid - 1

        return start
